runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c

```diff
@@ -158,6 +158,34 @@ static void genNumLocalesOptions(FILE* slurmFile, sbatchVersion sbatch,
   }
 }
 
+static void propagate_environment(FILE *f)
+{
+  char *lang = getenv("LANG");
+  char *lc_all = getenv("LC_ALL");
+  char *lc_collate = getenv("LC_COLLATE");
+
+  // If any of the relevant character set environment variables
+  // are set, replicate the state of all of them.
+  if (lang || lc_all || lc_collate) {
+    fprintf(f, " env");
+
+    // All the undefines have to come first.
+    if (!lang)
+      fprintf(f, " -u LANG");
+    if (!lc_all)
+      fprintf(f, " -u LC_ALL");
+    if (!lc_collate)
+      fprintf(f, " -u LC_COLLATE");
+
+    if (lang)
+      fprintf(f, " LANG=%s", lang);
```

Comment:
There's not the least chance any of these 3 env var settings will contain a character that's a meta-character to the shell that runs the gasnetrun_obv command, is there?  (I don't think so, but that may mostly reflect the limits of my own knowledge.)

Reply:
I thought of that, and added a commit that put quotes around those characters.  I had to take the quotes back out because metacharacters turn out not to be interpreted here, and it was making the quotes part of the string.

Reply:
Oh!  So the command must not be interpreted by a shell after all.  A follow-up: there's not the least chance that there will be a white space character within the values of any of these 3, is there?   :-)

Reply:
I guess someone could always define one of the environment variables incorrectly, but all the correct values have no special characters, including no spaces.  If they define them incorrectly, they have worse problems.